webpack/assets/javascripts/react_app/components/common/forms/NumericInput.js

```diff
@@ -1,48 +1,57 @@
-import NumericInput from 'react-numeric-input';
+import ReactNumericInput from 'react-numeric-input';
 import React from 'react';
 import PropTypes from 'prop-types';
 
 import { noop } from '../../../common/helpers';
 import CommonForm from './CommonForm';
 
-const TextInput = ({
+/**
+ * NumericInput field.
+ *
+ * === Change handler
+ * It does not support *onChange* prop as a change handler,
+ * as the value is original unparsed value and would confuse users.
+ * Use onValueChange instead with `valueAsNumber`, `valueAsString` and `input` attributes.
+ */
+const NumericInput = ({
   label,
   className,
   value,
-  onChange,
+  onValueChange,
   format,
   precision,
   minValue,
 }) => (
   <CommonForm label={label} className={`common-numericInput ${className}`}>
-    <NumericInput
+    <ReactNumericInput
       format={format}
       min={minValue}
       value={value}
       precision={precision}
-      onChange={onChange}
+      onChange={onValueChange}
     />
   </CommonForm>
 );
 
-TextInput.propTypes = {
+NumericInput.propTypes = {
   label: PropTypes.string,
   className: PropTypes.string,
   value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
   format: PropTypes.func,
   precision: PropTypes.number,
   minValue: PropTypes.number,
-  onChange: PropTypes.func,
+  /** Only supported change handler. */
+  onValueChange: PropTypes.func,
```

Comment:
@Laviro to followup the discussion, would it make sense like this?
As there is no reasonable way to introduce classical onChange (with input event as a parameter).
**Note:** We could throw `onChange` event by hacking underlying component, but its `.target.value` would give unformatted value so for example `5 Gb` and that could lead users to parse it by them self instead of using `onValueChange`.